### ISSUE TITLE
Fails to build with Lua 5.4

### ISSUE THREAD

**Issue by mcepl** [editorconfig/editorconfig-core-lua#2]
When updating lua on openSUSE to 5.4.0, building of this package failed:
```
[    2s] + /usr/bin/cmake /home/abuild/rpmbuild/BUILD/editorconfig-core-lua-0.3.0/. '-GUnix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DCMAKE_INSTALL_LIBDIR:PATH=/usr/lib64 -DCMAKE_INSTALL_LIBEXECDIR=/usr/lib -DCMAKE_BUILD_TYPE=RelWithDebInfo '-DCMAKE_C_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG' '-DCMAKE_CXX_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG' '-DCMAKE_Fortran_FLAGS=-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG' '-DCMAKE_EXE_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' '-DCMAKE_MODULE_LINKER_FLAGS=-flto=auto -Wl,--as-needed' '-DCMAKE_SHARED_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' -DLIB_SUFFIX=64 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON -DBUILD_STATIC_LIBS:BOOL=OFF -DCMAKE_COLOR_MAKEFILE:BOOL=OFF -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_MODULES_INSTALL_DIR=/usr/lib64/cmake/lua54-editorconfig-core-lua -DECL_LIBDIR:PATH=/usr/lib64/lua/5.4 -DLUA_LIBRARY:PATH=/usr/lib64/lua/5.4 -DLUA_INCLUDE_DIR:PATH=/usr/include/lua5.4
[    2s] -- The C compiler identification is GNU 10.1.1
[    2s] -- Check for working C compiler: /var/lib/build/ccache/bin/cc
[    2s] -- Check for working C compiler: /var/lib/build/ccache/bin/cc - works
[    2s] -- Detecting C compiler ABI info
[    2s] -- Detecting C compiler ABI info - done
[    2s] -- Detecting C compile features
[    2s] -- Detecting C compile features - done
[    2s] -- Found EditorConfig: /usr/lib64/libeditorconfig.so  
[    2s] CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
[    2s]   Could NOT find Lua (missing: LUA_INCLUDE_DIR) (Required is at least version
[    2s]   "5.2")
[    2s] Call Stack (most recent call first):
[    2s]   /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
[    2s]   /usr/share/cmake/Modules/FindLua.cmake:232 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
[    2s]   CMakeLists.txt:39 (find_package)
[    2s] 
[    2s] 
[    2s] -- Configuring incomplete, errors occurred!
```

[Complete build log with all details, versions of packages, etc.](https://github.com/editorconfig/editorconfig-core-lua/files/4892447/_log.txt)

**Comment by cxw42** [comment]
I'm sorry to say I haven't worked with this core at all --- has anyone else?  5.3->5.4 may require some code changes as well as the build fix ([reference](https://www.lua.org/manual/5.4/manual.html#8)).

**Comment by randstr** [comment]
Thanks for reminding me to look into this.

**Comment by randstr** [comment]
This is a cmake [bug](https://gitlab.kitware.com/cmake/cmake/-/merge_requests/4959). The fix was merged for 3.18.

I still need to find time to check for binary compatibility with 5.4.

**Comment by mcepl** [comment]
Yes, it at least builds with [the patched cmake](https://build.opensuse.org/request/show/819696).
